src/eXpansion/Framework/Core/Storage/GameDataStorage.php

```diff
@@ -116,7 +116,18 @@ public function getGameModeCode()
      */
     public function getTitle()
     {
-        return $this->titles->get($this->getVersion()->titleId, self::TITLE_UNKNOWN);
+
+        $title = $this->titles->get($this->getVersion()->titleId, self::TITLE_UNKNOWN);
+        if ($title == self::TITLE_UNKNOWN) {
+            if (substr($this->getVersion()->titleId, 0, 2) == "TM") {
+                return "TM";
+            }
+            if (substr($this->getVersion()->titleId, 0, 2) == "SM") {
+                return "SM";
+            }
+        }
```

Comment:
I am bit worried about this. 

The issue the titleId could be something like `mySuperSmPack@TMaddict` and it woul consider it as TM. That is why I had not "tried" to deduce title from the titleId. 

To think about before merging this.

Reply:
as you see, well it first matches the array with titleid then tries to find first 2 letter if they match with the string... and it's better than nothing :)

Reply:
so it would in your case match `my`

Reply:
ooo I read it wrong, sorry Ok for this then